src/backend/plugins/config-gnome/config-gnome.c

```diff
@@ -0,0 +1,248 @@
+/* config-gnome.c
+ *
+ * Copyright 2022-2023 The Libproxy Team
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include <libpeas/peas.h>
+
+#include "config-gnome.h"
+
+#include "px-plugin-config.h"
+#include "px-manager.h"
+
+struct _PxConfigGnome {
+  GObject parent_instance;
+  GSettings *proxy_settings;
+  GSettings *http_proxy_settings;
+  GSettings *https_proxy_settings;
+  GSettings *ftp_proxy_settings;
+  GSettings *socks_proxy_settings;
+  gboolean available;
+};
+
+typedef enum {
+  GNOME_PROXY_MODE_NONE,
+  GNOME_PROXY_MODE_MANUAL,
+  GNOME_PROXY_MODE_AUTO
+} GnomeProxyMode;
+
+static void px_config_iface_init (PxConfigInterface *iface);
+void peas_register_types (PeasObjectModule *module);
+
+G_DEFINE_FINAL_TYPE_WITH_CODE (PxConfigGnome,
+                               px_config_gnome,
+                               G_TYPE_OBJECT,
+                               G_IMPLEMENT_INTERFACE (PX_TYPE_CONFIG, px_config_iface_init))
+
+enum {
+  PROP_0,
+  PROP_CONFIG_OPTION
+};
+
+static void
+px_config_gnome_init (PxConfigGnome *self)
+{
+  GSettingsSchemaSource *source;
+  GSettingsSchema *proxy_schema;
+  const char *desktops;
+
+  self->available = FALSE;
+
+  desktops = getenv ("XDG_CURRENT_DESKTOP");
+  if (!desktops)
+    return;
+
+  /* Remember that XDG_CURRENT_DESKTOP is a list of strings. */
+  if (strstr (desktops, "GNOME") == NULL)
+    return;
+
+  source = g_settings_schema_source_get_default ();
+  if (!source)
+    return;
```

Comment:
This should proably trigger a warning: It detected GNOME, but important stuff is missing that should be there.